lib/stripe.rb

```diff
@@ -210,6 +212,38 @@ def self.max_network_retries=(val)
 
   private
 
+  # A "cipher list" to be passed down to OpenSSL. We use defaults mostly, but
+  # explicitly disallow old secure protocols.
+  #
+  #     https://wiki.openssl.org/index.php/Manual:Ciphers(1)#CIPHER_STRINGS
+  def self.ciphers
+    @ciphers ||= begin
+      # This code has been duplicated from rest-client, which is the underlying
+      # library we're using to make HTTP requests. Ruby >= 2.0 is considered to
+      # have a pretty good cipher list, but older versions may have had a weak
+      # default cipher list, and we can't guarantee those aren't out of use
+      # yet. For now, look for one of those weak default lists and if found,
+      # replace it with a good one that's bundled within rest-client.
+      #
+      # All of this can be removed once this library is no longer compatible
+      # with Ruby 1.9.
+      ciphers = OpenSSL::SSL::SSLContext::DEFAULT_PARAMS.fetch(:ciphers)
+      if RestClient::Request::WeakDefaultCiphers.include?(ciphers)
+        ciphers = RestClient::Request::DefaultCiphers
+      end
+
+      # We mutate the default list a bit so that we can disallow old versions
+      # of TLS. We (the payments space) is a little ahead of the curve here in
+      # that even though TLS 1.0/1.1 haven't been retired yet, we want to make
+      # sure we deprecate them by June 2018 for reasons of PCI compliance.
+      ciphers = ciphers.split(":")
+      ciphers = ["!SSLv2", "!SSLv3", "!TLSv1"]
```

Comment:
Err ... yes, hah. I typoed something during a refactor. Thanks.

I'll add a few test cases to make sure that what this produces looks semi-sane.